db/prefixeddb/prefixeddb.go

```diff
@@ -0,0 +1,124 @@
+package prefixeddb
+
+import (
+	"bytes"
+
+	"go.vocdoni.io/dvote/db"
+)
+
+// PrefixedDatabase wraps a db.Database prefixing all keys with `prefix`.
+type PrefixedDatabase struct {
+	prefix []byte
+	db     db.Database
+}
+
+// check that PrefixedDatabase implements the db.Database interface
+var _ db.Database = (*PrefixedDatabase)(nil)
+
+// NewPrefixedDatabase creates a new PrefixedDatabase.  If the db is already a
+// PrefixedDatabase, instead of wrapping again, the prefixes are appended to
+// avoid unecessay layers.
+func NewPrefixedDatabase(db db.Database, prefix []byte) *PrefixedDatabase {
+	if pdb, ok := db.(*PrefixedDatabase); ok {
+		return &PrefixedDatabase{append(pdb.prefix, prefix...), pdb.db}
```

Comment:
Here's an example: https://play.golang.org/p/e1Slm5Ai-ug

You'll see that the output sometimes has duplicates, because the goroutines "collide" when reusing the same backing array for the prefix slice, since it has enough capacity that the appends happen in-place:

```
foo04
foo06
foo05
foo01
foo09
foo07
foo08
foo02
foo03
foo03
```

This does not hapen if you fix the capacity to its length, so that all appends must copy.

Reply:
Thanks for spotting this!